src/cmd/linuxkit/cache/write.go

```diff
@@ -69,7 +69,25 @@ func (p *Provider) ImagePull(ref *reference.Spec, trustedRef, architecture strin
 	ii, err := desc.ImageIndex()
 	if err == nil {
 		log.Debugf("ImageWrite retrieved %s is index, saving", pullImageName)
-		err = p.cache.ReplaceIndex(ii, match.Name(image), layout.WithAnnotations(annotations))
```

Comment:
What you are doing makes sense, but it does make it harder to understand, as the `else` case looks completely different from the `if` case.

Would it be easier to just change `ReplaceIndex()` to support optional architectures, so you can do:

```go
p.cache.ReplaceIndex(ii, match.Name(image), layout.WithAnnotations(annotations), platforms)
```

with the latter being optional?

Reply:
Sorry, probably I miss something, but `p.cache.ReplaceIndex` is a library function, I cannot modify it.

Reply:
Sure you can. The library function is in a different directory inside linuxkit.

Reply:
This function come from `go-containerregistry`, do you want me to modify vendor directory? It sounds sub-optimal.

Reply:
Oh it does? I’d forgotten which we have and which came from there. Then best leave it.